internal/lsp/language_id.go

```diff
@@ -0,0 +1,30 @@
+package lsp
+
+import (
+	"fmt"
+	"strings"
+)
+
+// LanguageID represents the coding language
+// of a file
+type LanguageID string
+
+const (
+	Terraform LanguageID = "terraform"
+	Tfvars    LanguageID = "tfvars"
+	Unknown   LanguageID = "Unknown"
+)
+
+func (l LanguageID) String() string {
+	return string(l)
+}
+
+func LanguageIDFromPath(path string) (LanguageID, error) {
```

Comment:
AFAIK per LSP spec, it is client's responsibility to make assumptions about language IDs based on extensions. Server should accept any file/bytes it is given and languageID assigned to it and not make any further assumptions.

I understand that internally the LS does need to parse files other than the ones provided by the client (so that it has full module context), but that logic should be IMO contained locally and not signal any relationship with language ID as that's LSP concept. We can just compare the file extensions in the parser, like we do now for `*.tf`.